### PR TITLE
minify var_export on AbstractListener::writeArrayToFile()

### PR DESCRIPTION
**Pull request by samsonasik**
from http://php.net/manual/fr/function.var-export.php#54440
tested in https://3v4l.org/gX3kA

Do we need tests `file_put_contents` with vfsStream?

### CODE REVIEW

File: src/Listener/AbstractListener.php [hunk 1]
```diff
@@ -59,10 +59,31 @@ public function setOptions(ListenerOptions $options)
...
+     * @return string
+     */
+    private function varExportMin($var)
+    {
+        if (is_array($var)) {
+            $toImplode = [];
+            foreach ($var as $key => $value) {
+                $toImplode[] = var_export($key, true) . '=>' . $this->varExportMin($value, true);
+            }
+            return '['.implode(',', $toImplode).']';
```
Comment (vaclavvanik) [inline src/Listener/AbstractListener.php]: Can you add spaces before and after `.`?

Comment (samsonasik) [inline src/Listener/AbstractListener.php]: done

**Comment by weierophinney** [comment]
What does this accomplish, exactly? What problem does it solve?

**Comment by samsonasik** [comment]
for performance, minified should be faster I think

**Comment by weierophinney** [comment]
> for performance, minified should be faster I think

I'd like to see benchmarks, then. PHP code generally doesn't benefit from minimization efforts...

File: src/Listener/AbstractListener.php [hunk 2]
```diff
@@ -59,10 +59,31 @@ public function setOptions(ListenerOptions $options)
...
+     *
+     * @param  string|array $var
+     * @return string
+     */
+    private function varExportMin($var)
+    {
+        if (is_array($var)) {
+            $toImplode = [];
+            foreach ($var as $key => $value) {
+                $toImplode[] = var_export($key, true) . '=>' . $this->varExportMin($value, true);
```
Comment (webimpress) [inline src/Listener/AbstractListener.php]: As I can see `varExportMin` has only one parameter, here we have call with two params.

**Comment by samsonasik** [comment]
@weierophinney do you mean something a sample like in  https://github.com/zendframework/zend-servicemanager/tree/master/benchmarks ?

File: src/Listener/AbstractListener.php (hunk 2 above)
Comment (samsonasik) [inline src/Listener/AbstractListener.php]: fixed

**Comment by Ocramius** [comment]
Dito, plus `var_export` is a potential source of security issues if
replaced with a custom solution.

**Comment by weierophinney** [comment]
Since @Ocramius brings up security implications, I'm going to close this.

**Pull request closed without merging**